leukeleu_django_gdpr/anonymize.py

```diff
@@ -0,0 +1,140 @@
+from functools import partial
+
+from faker import Faker
+
+from django.apps import apps
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+from django.db.models import Q
+
+from leukeleu_django_gdpr.gdpr import read_data
+
+
+def get_models_from_gdpr_yml():
+    data = read_data()
+    return data["models"]
+
+
+class BaseAnonymizer:
+    excluded_fields = []
+    extra_fieldtype_overrides = None
+    extra_qs_overrides = None
+    extra_field_overrides = None
+
+    def __init__(self):
+        self.fake = Faker(["nl-NL"])
+
+    def anonymize(self):
+        fieldtype_overrides = self.get_fieldtype_overrides()
+        qs_overrides = self.get_qs_overrides()
+        field_overrides = self.get_field_overrides()
+
+        with transaction.atomic():
+            models = get_models_from_gdpr_yml()
+            for model_name, model_data in models.items():
+                print(f"Currently anonymizing: {model_name}")
+
+                Model = apps.get_model(model_name)  # noqa: N806
+
+                qs = qs_overrides.get(model_name, Model._base_manager.all())
+                qs = qs.all()  # Makes sure we are always dealing with the latest data
+
+                for field_name, field_data in model_data["fields"].items():
+                    field_path = f"{model_name}.{field_name}"
+                    if not field_data["pii"] or field_path in self.excluded_fields:
+                        # Leave non PII and ignored fields alone
+                        continue
+
+                    field = Model._meta.get_field(field_name)
+
+                    field_type = type(field).__name__
+                    if field.unique:
+                        field_type += ".unique"
+
+                    try:
+                        value_func = field_overrides.get(
+                            field_path,
+                            fieldtype_overrides[field_type],
+                        )
+                    except KeyError:
+                        raise ImproperlyConfigured(
+                            f"Field type '{field_type}' not defined "
+                            "inside FIELDTYPE_FAKER_MAPPING"
+                        )
+
+                    for obj in qs:
+                        setattr(obj, field_name, value_func())
```

Comment:
Misschien hier "empty" values met rust laten? Of dus checken in condition in L46?